src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlCommand.cs

```diff
@@ -376,14 +373,6 @@ private SqlCommand(SqlCommand from) : this()
                     }
                 }
 
-                // Check to see if the currently set transaction has completed.  If so,
```

Comment:
Behavior change. I have a command lying around and the transaction was changed for the command. Now I associate the new connection with it, and it magically overwrote previous transaction. 

Can we stick to the lowest common denominator on how this convenience of deriving the transaction for this command is going to work, while preserving most of the old behavior ?

Reply:
> @saurabh500  Behavior change. I have a command lying around and the transaction was changed for the command. Now I associate the new connection with it, and it magically overwrote previous transaction.

Since a transaction's lifetime is `BeginTransaction - Commit/Rollback/throw`, and a single connection has a single active transaction, how can a command be lying around with a different transaction that magically changes? If it had a different transaction reference associated, and the connection ran another transaction, then it's a complete transaction which used to be nullified internally in the old functionality too.

If you associate a new connection to it, you don't do it in the hopes of keeping the old transaction on the `.Transaction` property and a different connection in the `.Connection`. This would not have passed validation previously also.

A few lines of code explaining this behavior would be helpful.

Reply:
@saurabh500 

Could you provide a few lines of code to verify how these code changes would break?